invenio_administration/assets/semantic-ui/js/invenio_administration/details/AdminDetailsView.js

```diff
@@ -0,0 +1,131 @@
+import DetailsComponent from "./DetailsComponent";
+import DetailsActions from "./DetailsActions";
+import { InvenioAdministrationActionsApi } from "../api/actions";
+import AdminDetailsContext from "./AdminDetailsContext";
+
+export default class AdminDetailsView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      data: undefined,
+      error: undefined,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = async () => {
+    this.setState({ loading: true });
+    const { apiEndpoint, pid } = this.props;
+
+    try {
+      const response = await InvenioAdministrationActionsApi.getResource(
+        apiEndpoint,
+        pid
+      );
+
+      this.setState({
+        loading: false,
+        data: response.data,
+        error: undefined,
+      });
+    } catch (e) {
+      console.error(e);
+      //TODO
+    }
+  };
+
+  /**
+   * Sorts given columns given their provided order.
+   * Returns an array of objects, each object being composed as:
+   * {
+   *   key: str,
+   *   text: str,
+   *   order: number
+   * }
+   *
+   * @param {Object[]} columns
+   * @returns {Object[]} an array of objects, ordered by the provided order.
+   */
+  sortColumns = (columns) => {
+    let sorted = [];
+    Object.keys(columns).forEach((key) => {
+      let innerObj = columns[key];
+      if (
+        Object.prototype.hasOwnProperty.call(innerObj, "text") &&
+        Object.prototype.hasOwnProperty.call(innerObj, "order")
+      ) {
+        sorted.push({
+          key: key,
+          text: columns[key].text,
+          order: innerObj.order,
+        });
+      }
+    });
+    return sorted.sort((a, b) => a.order > b.order);
+  };
+
+  render() {
+    const { title, columns, actions, displayEdit, displayDelete, children } =
+      this.props;
+    const { loading, data } = this.state;
+
+    const sortedColumns = this.sortColumns(columns);
+    return (
+      <Grid stackable>
+        <Grid.Row columns="2">
+          <Grid.Column verticalAlign="middle">
+            <Header as="h1" size="large">
+              {title}
+            </Header>
+          </Grid.Column>
+
+          <Grid.Column verticalAlign="middle" floated="right" textAlign="right">
+            <DetailsActions
+              actions={actions}
+              displayEdit={displayEdit}
+              displayDelete={displayDelete}
+              details={data}
+            />
+          </Grid.Column>
+        </Grid.Row>
+        <Grid.Column width="16">
+          {loading ? (
+            <Segment className="rel-pt-4 rel-pb-4">
+              <Dimmer active inverted>
+                <Loader />
+              </Dimmer>
+            </Segment>
+          ) : (
+            <AdminDetailsContext.Provider value={{ data: data }}>
+              <DetailsComponent {...this.props} columns={sortedColumns} />
+              {children}
```

Comment:
Added composition because it allows other components to extend the details page without overriding the table. 
Sometimes overriding the whole component is not what is wanted, e.g. in OAI-PMH. In that case, the modules adds another table to the bottom of the page.